pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -930,3 +956,47 @@ func GetSelectorRegexMap(l7 policy.L7DataMap) (CachedSelectorREEntry, error) {
 
 	return newRE, nil
 }
+
+// Removes the last trailing dot in case its an fqdn
+// TODO move to the proper location
+func FromFqdn(s string) string {
+	if !dns.IsFqdn(s) {
+		return s
+	}
+	return s[:len(s)-1]
+}
+
+// Reverse reverses a string
+// TODO figure out the best possible way to do this with minimal overhead
+func Reverse(s string) string {
+	size := len(s)
+	buf := make([]byte, size)
+	for start := 0; start < size; {
+		r, n := utf8.DecodeRuneInString(s[start:])
+		start += n
+		utf8.EncodeRune(buf[size-start:], r)
+	}
+	return string(buf)
+}
```

Comment:
Would be good to write a benchmark test for this function so that we can evaluate other implementations and compare them.